src/Symfony/Component/Translation/DoctrineMessageCache.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation;
+
+use Doctrine\Common\Cache\Cache;
+
+/**
+ * @author Abdellatif Ait Boudad <a.aitboudad@gmail.com>
+ */
+class DoctrineMessageCache implements MessageCacheInterface
+{
+    const CACHE_RESOURCE_HASH = 'resources_hash';
+    const CATALOGUE_FALLBACK_LOCALE = 'fallback_locale';
+
+    /**
+     * @var bool
+     */
+    private $debug;
+
+    /**
+     * @var Cache
+     */
+    private $cacheProvider;
+
+    /**
+     * @param Cache $cacheProvider
+     * @param bool  $debug
+     */
+    public function __construct(Cache $cacheProvider, $debug = false)
+    {
+        $this->debug = $debug;
+        $this->cacheProvider = $cacheProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isFresh($locale, $options = array())
+    {
+        if (!isset($options['resources_hash'])) {
+            $options['resources_hash'] = '';
```

Comment:
instead of altering the array of options to add the key just to be able to read it later, I would rather use a local variable for the expected resource hash.
this will avoid copying the array because of writing in it